agent/bench-scripts/postprocess/process-iteration-samples

```diff
@@ -188,13 +195,13 @@ foreach $sample (@json_samples) { # samples 0..N
 		# array, but it's possible to have more (for a potential multi-benchmark-type workload)
 		foreach my $metric_name (sort keys %{ $sample_hash{$metric_type} }) {
 			for (my $i = 0; $i < scalar @{ $sample_hash{$metric_type}{$metric_name} }; $i++) {
-			my $mapped_uid = get_uid($sample_hash{$metric_type}{$metric_name}[$i]{get_label('uid_label')}, \%{ $sample_hash{$metric_type}{$metric_name}[$i] });
+				my $mapped_uid = get_uid($sample_hash{$metric_type}{$metric_name}[$i]{get_label('uid_label')}, \%{ $sample_hash{$metric_type}{$metric_name}[$i] });
 				if ($mapped_uid) {
 					# either find metric_name element within %iteration with same uid, or create a new one
 					my $iteration_matching_index;
-					if ($iteration{$metric_type}{$metric_name}) {
-						for (my $j = 0; $j < scalar @{$iteration{$metric_type}{$metric_name}}; $j++) {
-							if ( $iteration{$metric_type}{$metric_name}[$j]{get_label('uid_label')} )  {
+					if (exists($iteration{$metric_type}{$metric_name})) {
+						for (my $j = 0; $j < scalar @{ $iteration{$metric_type}{$metric_name} }; $j++) {
+							if (exists($iteration{$metric_type}{$metric_name}[$j]{get_label('uid_label')}))  {
```

Comment:
I don't know, I believe that historically we wanted to describe how the UID is generated, so that it gets generated the same way each time it is needed, and I think the intention was to avoid having a UID that is generated incorrectly where the description for how it is generated doesn't match.

I have never been in favor of this method as it is compute intensive (on the pbench-server side, we generate the UID once, letting the code hold the description of how it is generated).

This is not something that I would want to change as part of this PR.  If we really want to change this, we should file an issue.